### PR TITLE
implement network_interface from netdev_stdlib

### PR DESCRIPTION
**Pull request by tphoney**

### CODE REVIEW

File: lib/puppet/provider/network_interface/nxapi.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+Puppet::Type.type(:network_interface).provide(:nxapi, parent: Puppet::Type.type(:cisco_interface).provider(:nxapi)) do
+  @doc = 'network INTERFACE'
+
+  has_features :activable, :describable
```
Comment (mikewiebe) [inline lib/puppet/provider/network_interface/nxapi.rb]: Where are these features defined?

**Comment by mikewiebe** [comment]
Standard Items:

1) Please update the CHANGELOG.md
2) Update the resource reference. 
  * https://github.com/cisco/cisco-network-puppet-module#resource-reference.
  * You can follow the exampes for netdev_stdlib here: https://github.com/cisco/cisco-network-puppet-module/blob/release_1.1.0/README.md#resource-reference
3) Update examples: https://github.com/cisco/cisco-network-puppet-module/blob/develop/examples/demo_interface.pp
4) Include beaker test results.

**Comment by tphoney** [comment]
Resource :: All Attributes Defaults :: End
tests/beaker_tests/netdev_stdlib/network_interface_provider_defaults.rb passed in 23.78 seconds
      Test Suite: tests @ 2015-11-04 15:37:51 +0000

      - Host Configuration Summary -


              - Test Case Summary for suite 'tests' -
       Total Suite Time: 23.78 seconds
      Average Test Time: 23.78 seconds
              Attempted: 1
                 Passed: 1
                 Failed: 0
                Errored: 0
                Skipped: 0
                Pending: 0
                  Total: 1

      - Specific Test Case Status -

Failed Tests Cases:
Errored Tests Cases:
Skipped Tests Cases:
Pending Tests Cases:

File: lib/puppet/provider/network_interface/nxapi.rb (hunk 1 above)
Comment (tphoney) [inline lib/puppet/provider/network_interface/nxapi.rb]: These features are part of the netdev_stdlib https://github.com/puppetlabs/netdev_stdlib

**Comment by tphoney** [comment]
@mikewiebe i think that is everything, the changelog is already updated

File: lib/puppet/provider/network_interface/nxapi.rb (hunk 1 above)
Comment (mikewiebe) [inline lib/puppet/provider/network_interface/nxapi.rb]: Ok.. I found :describable but it's under network_vlan? 

https://github.com/puppetlabs/netdev_stdlib/blob/f7184a30bed58635fc7e1a226f0a64530186ab5d/lib/puppet/type/network_vlan.rb#L9

I could not find :activable.

File: README.md [hunk 2]
```diff
@@ -202,6 +203,7 @@ cisco_interface_ospf {"Ethernet1/2 Sample":
 * [`cisco_vlan`](#type-cisco_vlan)
 * [`cisco_vrf`](#type-cisco_vrf)
 * [`cisco_vtp`](#type-cisco_vtp)
+* [`network_interface`](#type-network_interface)
```
Comment (mikewiebe) [inline README.md]: IMO, this should be under Interface Types and we are adding the netdev_stdlib tag.

 * \[`network_interface (netdev_stdlib)`\](#type-network_interface)

**Comment by tphoney** [comment]
All Attributes Defaults :: End
tests/beaker_tests/netdev_stdlib/network_interface_provider_defaults.rb passed in 22.38 seconds
      Test Suite: tests @ 2015-11-04 16:35:48 +0000

      - Host Configuration Summary -


              - Test Case Summary for suite 'tests' -
       Total Suite Time: 22.38 seconds
      Average Test Time: 22.38 seconds
              Attempted: 1
                 Passed: 1
                 Failed: 0
                Errored: 0
                Skipped: 0
                Pending: 0
                  Total: 1

  1 implement network_interface from netdev_stdlib
      - Specific Test Case Status -

File: README.md (hunk 2 above)
Comment (tphoney) [inline README.md]: does that replace line 206  ? @mikewiebe

Comment (mikewiebe) [inline README.md]: @tphoney It replaces line 160.  Sorry for the confusion.

File: tests/beaker_tests/netdev_stdlib/network_interfacelib.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+# A library to assist testing network_interface resource
+module NetworkInterfaceLib
+  # A. Methods to create manifests for network_interface Puppet provider test cases.
+
+  def self.create_defaults
+    manifest_str = "cat <<EOF >#{UtilityLib::PUPPETMASTER_MANIFESTPATH}
+node default {
+    network_interface { 'ethernet1/4':
+      speed                        => 'auto',
```
Comment (mikewiebe) [inline tests/beaker_tests/netdev_stdlib/network_interfacelib.rb]: These should be set to 'default' instead right?  Also, you are missing description.  It won't show up in the output, but we should still exercise the underlying logic.

Comment (mikewiebe) [inline tests/beaker_tests/netdev_stdlib/network_interfacelib.rb]: @tphoney Never mind... I see that we don't support 'default' for these.

Comment (tphoney) [inline tests/beaker_tests/netdev_stdlib/network_interfacelib.rb]: @mikewiebe  the netdev_stdlib network_interface type does not have a value of default.

**Comment by tphoney** [comment]
readme updated as well @mikewiebe

**Comment by mikewiebe** [comment]
:+1:

**Pull request merged**